keras/preprocessing/image.py

```diff
@@ -563,14 +567,15 @@ def __init__(self, directory, image_data_generator,
 
         for subdir in classes:
             subpath = os.path.join(directory, subdir)
-            for fname in sorted(os.listdir(subpath)):
-                is_valid = False
-                for extension in white_list_formats:
-                    if fname.lower().endswith('.' + extension):
-                        is_valid = True
-                        break
-                if is_valid:
-                    self.nb_sample += 1
+            for root, dirs, files in os.walk(subpath, followlinks=follow_links):
```

Comment:
if you mean the `followlinks`, `follow_links` difference, I wanted to match the underscore convention in Keras. I realize it collides with `os.walk`.